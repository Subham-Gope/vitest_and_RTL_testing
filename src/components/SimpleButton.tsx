import React from "react";

const SimpleButton: React.FC = () => {
  return (
    <>
      <button>Click Me</button>
    </>
  );
};

export default SimpleButton;
