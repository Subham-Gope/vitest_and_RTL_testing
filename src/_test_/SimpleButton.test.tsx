import { test, describe, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import SimpleButton from "../components/SimpleButton";

describe("Testing the SimpleButton.tsx file", () => {
  test("testing the component to return 'Click Me' ", () => {
    const { container } = render(<SimpleButton />);

    expect(container).toMatchSnapshot();
    const message = screen.getByText("Click Me");
    expect(message).toBeInTheDocument();
  });
});
