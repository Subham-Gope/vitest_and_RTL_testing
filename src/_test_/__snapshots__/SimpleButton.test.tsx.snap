// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the SimpleButton.tsx file > testing the component to return 'Click Me'  1`] = `
<div>
  <button>
    Click Me
  </button>
</div>
`;
